packages/react/src/transactions/adapters/useCreateCommentController.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  PendingSigningRequestError,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import { CreateComment, CreateCommentRequest } from '@lens-protocol/domain/use-cases/publications';
+import { ProtocolCallUseCase } from '@lens-protocol/domain/use-cases/transactions';
+
+import { useSharedDependencies } from '../../shared';
+import { PromiseResultPresenter } from './PromiseResultPresenter';
+import { UploadHandler } from './UploadHandler';
+import { CommentCallGateway } from './publication-call-gateways/CommentCallGateway';
+
+export type UseCreateCommentControllerArgs = {
+  upload: UploadHandler;
+};
+
+export function useCreateCommentController({ upload }: UseCreateCommentControllerArgs) {
+  const {
+    apolloClient,
+    transactionFactory,
+    activeWallet,
+    transactionGateway,
+    protocolCallRelayer,
+    transactionQueue,
+  } = useSharedDependencies();
+
+  return async (request: CreateCommentRequest) => {
+    const gateway = new CommentCallGateway(apolloClient, transactionFactory, upload);
```

Comment:
Totally agree in deferring the creation of gateway now that is no longer a shared resource.